flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesClusterDescriptor.java

```diff
@@ -112,18 +112,27 @@ public String getClusterDescription() {
                 return new RestClusterClient<>(
                         configuration,
                         clusterId,
-                        new StandaloneClientHAServices(
-                                HighAvailabilityServicesUtils.getWebMonitorAddress(
-                                        configuration,
-                                        HighAvailabilityServicesUtils.AddressResolution
-                                                .TRY_ADDRESS_RESOLUTION)));
+                        new StandaloneClientHAServices(getWebMonitorAddress(configuration)));
             } catch (Exception e) {
                 throw new RuntimeException(
                         new ClusterRetrieveException("Could not create the RestClusterClient.", e));
             }
         };
     }
 
+    private String getWebMonitorAddress(Configuration configuration) throws Exception {
+        HighAvailabilityServicesUtils.AddressResolution resolution =
+                HighAvailabilityServicesUtils.AddressResolution.TRY_ADDRESS_RESOLUTION;
+        if (configuration.get(KubernetesConfigOptions.REST_SERVICE_EXPOSED_TYPE)
+                == KubernetesConfigOptions.ServiceExposedType.ClusterIP) {
+            resolution = HighAvailabilityServicesUtils.AddressResolution.NO_ADDRESS_RESOLUTION;
+            LOG.warn(
+                    "Please note that Flink client operation(e.g. cancel, list, stop,"
+                            + " savepoint, etc.) won't work from outside the Kubernetes cluster.");
```

Comment:
Let's add that this is due to having chosen `ClusterIP` for the `KubernetesConfigOptions.REST_SERVICE_EXPOSED_TYPE`

Reply:
This log only shows up when the having chosen `ClusterIP`. So I do not add this information here.

But I also think it is harmless to show more information.